src/components/form/add-more-section.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { ReactElement, useState } from "react";
+import { Button } from "../button";
+
+type ChildProps = {
+  index: number;
+  onDelete: (index:number) => void;
+  elementId: number;
+}
+
+type Props = {
+  buttonText?: string;
+  className?: string;
+  children?: ReactElement<ChildProps>;
+};
+
+// The section will accept an element and dupclicate upon button press. It also passes an index numbber and delete function as props to the element.
+export const AddMoreSection = ({ buttonText, className, children, ...rest }: Props) => {
+  const [childList, setChildList] = useState<number[]>([1]);
+  const [childIndex, setChildIndex] = useState<number>(2);
+
+  const addChild = () => {
+    setChildList((prevState) => [...prevState, childIndex]);
+    setChildIndex(childIndex + 1);
+  }
+
+  const removeChild = (index: number) => {
+    setChildList((prevState) => {
+      return prevState.filter((item) => item != index);
+    })
+  }
+
+  return (
+    <>
+    {childList.map((child, i) => {
+      const index = i + 1;
+
+      return (
+        <div key={child}>
+          {React.isValidElement(children)
+          ? React.cloneElement(children, { index, onDelete: removeChild, elementId: child })
```

Comment:
nice use of React's cloneElement! this seems to be a good use-case for that function. the React docs do mention that this function can lead to "fragile code" but perhaps having the isValidElement check beforehand will help with this?